# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/transportbridge/MediaPipelineFilter.cpp
# Commit: 5288094fbef3
# Full Hash: 5288094fbef3f1d0e895c65bebc2f2b33d7b2bfb
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - Fixes to MediaPipelineFilter. r=ng
#   
#   Upstream's implementation for rid and mid assumes they are always present but
#   the current code in MediaPipelineFilter had them optional. We need to double
#   check which way it should be.
# ==============================================================================

diff -r 0aa9398df7c0 -r 5288094fbef3 dom/media/webrtc/transportbridge/MediaPipelineFilter.cpp
--- a/dom/media/webrtc/transportbridge/MediaPipelineFilter.cpp	Mon Nov 09 15:03:01 2020 -0500
+++ b/dom/media/webrtc/transportbridge/MediaPipelineFilter.cpp	Mon Nov 09 15:26:46 2020 -0500
@@ -9,7 +9,7 @@
 
 #include "MediaPipelineFilter.h"
 
-#include "api/rtpparameters.h"
+#include "api/rtp_parameters.h"
 #include "mozilla/Logging.h"
 
 // defined in MediaPipeline.cpp
@@ -35,16 +35,11 @@
 bool MediaPipelineFilter::Filter(const webrtc::RTPHeader& header) {
   DEBUG_LOG(("MediaPipelineFilter inspecting seq# %u SSRC: %u",
              header.sequenceNumber, header.ssrc));
-
-  auto fromStreamId = [&](const webrtc::StreamId& aId) {
-    return Maybe<std::string>(aId.empty() ? Nothing() : Some(aId.data()));
-  };
-
   //
   //  MID Based Filtering
   //
 
-  const auto& mid = fromStreamId(header.extension.mid);
+  const auto& mid = Some(header.extension.mid);
 
   // Check to see if a bound SSRC is moved to a new MID
   if (mRemoteMidBindings.count(header.ssrc) == 1 && mid && mRemoteMid != mid) {